cmd/kaspawallet/daemon/server/bump_fee.go

```diff
@@ -0,0 +1,138 @@
+package server
+
+import (
+	"context"
+
+	"github.com/kaspanet/kaspad/app/appmessage"
+	"github.com/kaspanet/kaspad/cmd/kaspawallet/daemon/pb"
+	"github.com/kaspanet/kaspad/cmd/kaspawallet/libkaspawallet"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/txscript"
+	"github.com/pkg/errors"
+)
+
+func (s *server) BumpFee(_ context.Context, request *pb.BumpFeeRequest) (*pb.BumpFeeResponse, error) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	entry, err := s.rpcClient.GetMempoolEntry(request.TxID, false, false)
+	if err != nil {
+		return nil, err
+	}
+
+	domainTx, err := appmessage.RPCTransactionToDomainTransaction(entry.Entry.Transaction)
+	if err != nil {
+		return nil, err
+	}
+
+	mass := s.txMassCalculator.CalculateTransactionMass(domainTx) // TODO: Does GetMempoolEntry already provide the mass?
+	feeRate := float64(entry.Entry.Fee) / float64(mass)
+	newFeeRate, err := s.calculateFeeRate(request.FeeRate)
+	if err != nil {
+		return nil, err
+	}
+
+	if feeRate >= newFeeRate {
+		return nil, errors.Errorf("new fee rate (%f) is not higher than the current fee rate (%f)", newFeeRate, feeRate)
+	}
+
+	outpointsSet := make(map[externalapi.DomainOutpoint]struct{})
+	for _, input := range domainTx.Inputs {
+		outpointsSet[input.PreviousOutpoint] = struct{}{}
+	}
+
+	var maxUTXO *walletUTXO
+	for _, utxo := range s.utxosSortedByAmount {
```

Comment:
Afaiu `s.utxosSortedByAmount` will not hold current mempool entries (filtered [here](https://github.com/someone235/kaspad/blob/6901e7d53863ac5b7b49e03f018d256bee9bfd64/cmd/kaspawallet/daemon/server/sync.go#L234)), so it seems this search will most likely fail. 

I think we need to keep the mempool entries (during the refresh; in `s.excludedMempoolUtxos`), and search through them here as well